src/etc/inc/nat64.inc

```diff
@@ -0,0 +1,164 @@
+<?php
+/*
+ * nat64.inc
+ *
+ * part of pfSense (https://www.pfsense.org)
+ * Copyright (c) 2004-2013 BSD Perimeter
+ * Copyright (c) 2013-2016 Electric Sheep Fencing
+ * Copyright (c) 2014-2020 Rubicon Communications, LLC (Netgate)
+ * Copyright (c) 2020 Nick Whaley
+ * All rights reserved.
+ *
+ * originally part of m0n0wall (http://m0n0.ch/wall)
+ * Copyright (c) 2003-2006 Manuel Kasper <mk@neon1.net>.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+require_once("globals.inc");
+require_once("functions.inc");
+require_once("util.inc");
+require_once("notices.inc");
+
+function nat64_load_modules() {
+	mute_kernel_msgs();
+	
+	if (!is_module_loaded("ipfw.ko"))
+		mwexec("/sbin/kldload ipfw");
+	if (!is_module_loaded("ipfw_nat64.ko"))
+		mwexec("/sbin/kldload ipfw_nat64");
+	set_sysctl(array(
+		//having ipfw enabled in pfil conflicts with capture portal.
+		"net.inet.ip.pfil.inbound" => "pf,ipfw", //has ipfw for NAT44+NAT64
+		"net.inet6.ip6.pfil.inbound" => "pf",
+		"net.inet.ip.pfil.outbound" => "pf",
+		"net.inet6.ip6.pfil.outbound" => "pf",
+	    "net.link.ether.ipfw" => "1",
+	    "net.inet.ip.fw.nat64_direct_output" => "0",
+	    "net.inet.ip.fw.one_pass" => "1",
+	    "net.inet.ip.fw.tables_max" => "65534"
+	));
+
+	unmute_kernel_msgs();
+}
+
+function filter_generate_nat64_rules() {
+	global $config, $g;
+
+	init_config_arr(array('nat', 'nat64', 'rule'));
+	$a_nat64 = &$config['nat']['nat64']['rule'];
+
+	if (count($a_nat64) > $g['nat64_rules_interval'])
+		nat64_syslog("Out of space for nat64 rules. Some may be ignored.");
```

Comment:
pfSense style: Always use brackets on conditionals, even if it's a single sentence inside it
```php
if (foo) {
    bar();
}
```